mmedit/models/video_interpolater/basic_interpolater.py

```diff
@@ -0,0 +1,308 @@
+                                           f'{i:08d}-{iteration + 1:06d}.png')
+                elif iteration is None:
+                    save_path_i = osp.join(save_path, folder_name,
+                                           f'{i:08d}.png')
+                else:
+                    raise ValueError('iteration should be number or None, '
+                                     f'but got {type(iteration)}')
+                mmcv.imwrite(tensor2img(output[:, i, :, :, :]), save_path_i)
+
+    def forward_dummy(self, img):
+        """Used for computing network FLOPs.
+
+        Args:
+            img (Tensor): Input frames.
+
+        Returns:
+            Tensor: Output frame(s).
+        """
+        out = self.generator(img)
+        return out
+
+    def train_step(self, data_batch, optimizer):
+        """Train step.
+
+        Args:
+            data_batch (dict): A batch of data.
+            optimizer (obj): Optimizer.
+
+        Returns:
+            dict: Returned output.
+        """
+        outputs = self(**data_batch, test_mode=False)
+        loss, log_vars = self.parse_losses(outputs.pop('losses'))
+
+        # optimize
+        optimizer['generator'].zero_grad()
+        loss.backward()
+        optimizer['generator'].step()
+
+        outputs.update({'log_vars': log_vars})
+        return outputs
+
+    def val_step(self, data_batch, **kwargs):
+        """Validation step.
+
+        Args:
+            data_batch (dict): A batch of data.
+            kwargs (dict): Other arguments for ``val_step``.
+
+        Returns:
+            dict: Returned output.
+        """
+        output = self.forward_test(**data_batch, **kwargs)
+        return output
+
+    @staticmethod
+    def split_frames(input_tensors):
+        """split input tensors for inference.
+
+        This is a basic function, interpolate a frame between the given two
+        frames.
+
+        Args:
+            input_tensors (Tensor): Tensor of input frames.
+
+        Returns:
+            list[Tensor]: Splided tensors.
+        """
+
+        num_frames = input_tensors.shape[0]
+
+        result = []
+        for i in range(0, num_frames - 1, 1):
+            result.append(input_tensors[i:i + 2])
+
+        return result
+
+    @staticmethod
+    def merge_frames(input_images, output_images):
+        """merge input frames and output frames.
+
+        This is a basic function, interpolate a frame between the given two
+        frames.
+
+        Args:
+            input_images (list[np.array]): The input frames.
+            output_images (list[np.array]): The output frames.
+
+        Returns:
+            list[np.array]: The final frames.
+                in_frame, out_frame, in_frame, out_frame, in_frame ...
+        """
+
+        result = []
+        for i in range(len(output_images)):
+            result.append(input_images[i])
+            result.append(output_images[i])
+        result.append(input_images[-1])
```

Comment:
Sorry, I just find one use case of using Tensor in this function. Consider the case when a model takes two frames as input and interpolates 3 frames (t=0.25, 0.5, 0.75). In this case, the `input_images` should be 
[x0, x1, x2] and `output_images` is [y0, y1], where 
```
y0 = torch.stack([x0.25, x0.5, x0.75], dim=1)
y1 = torch.stack([x1.25, x1.5, x1.75], dim=1)
```
It would be easier if `input_images` and `output_images` are Tensors and we transform them to image here:
```
result = []
for i, output in enumerate(output_images):
    result.append(tensor2img(input_images[i]))
    for j in range(0, output.size(1)):
        result.append(tensor2img(output[:, j, :, :, :])
result.append(input_images[-1])

return result
```